Added via upoad - changing JS adn CSS code

CarballoDa opened the pull request:
The objective for this branch is to change the code :

**From JavaScript to Typescript :** 
- list.class.ts Done!. Compiled to list.class.js
- typology-list.ts work in progress.  Complied to typology-list.js

**From Bootstrap 5 to Tailwind CSS**
1. index.html pending

gtrias commented on public/js/list.class.js:
```diff
@@ -1,18 +1,103 @@
-
-    addTitle(value){
-        this.title = value;
+"use strict";
```
Ideally, you should not commit compiled js to git :) 

I would add them to `gitignore` or compile them into a different folder (maybe `dist`?) so it's easy to deal with this.
After compilation this js files are the ones you will deploy to final server. But this is not really interesting to have in git repository

CarballoDa replied:
Ok. I think this files will be added to git ignore. Tanks for the advice!

gtrias commented on public/js/list.class.ts:
```diff
@@ -0,0 +1,112 @@
+export class List{
```
Missing space after List. Have you considered using a formatter such as [prettier](https://prettier.io)?

CarballoDa replied:
I didn't use prettier ever. I will test it! Thanks!

gtrias commented on public/js/typology-list.ts:
```diff
@@ -0,0 +1,55 @@
+import { List } from './list.class';
+declare global { var list: List }
+
+list = new List();
```
`const list = new List()`

CarballoDa replied:
List has no type declaration in line 4 because has been declared as global var in line 2. ¿Is this incorrect?  
But thinking about, it is not necessary to declare this var as global. Removing line 2 and adding const to list in line 4 will work perfectly.

gtrias commented on public/js/typology-list.ts:
```diff
@@ -0,0 +1,55 @@
+* param {a} bool
+* param {b} bool
+*/
+function ableOrDisableListTools(a: boolean, b: boolean): void
```
Why do you need to pass two booleans here?

CarballoDa replied:
Your are right. I made the change with only one :)

gtrias commented:
Overall is good :) 
I think I would personally deal with Typescript in an individual PR, and then I would tackle Tailwind.

gtrias commented:
Why you ended up closing this PR?

Some nitpicks:
- Set a better branch name. `main-1` is not self-explanatory.
- Set better commit messages. A commit message should explain what is about instead on how it was created...

Maybe this will help you to have better git habits (at least to be inspired): https://www.conventionalcommits.org/

CarballoDa commented:
Ey! I closed this for various reasons, one of them the branch name (I didn't like main-1 as branch name). In order to change a branch name I must delete the branch and create it again. Besides I added .gitinore file to ignore al js files and moved .ts files into a new directory called /ts.
A new brach was created yesterday but I didn't want to bother you only for this.

In any case, I'm sending you the new Pull Request link : https://github.com/CarballoDa/typology-list/pull/2

Thank you for the git habits advice!
